lib/auth/auth.go

```diff
@@ -55,9 +55,10 @@ type Authority interface {
 	// GetNewKeyPairFromPool returns new keypair from pre-generated in memory pool
 	GetNewKeyPairFromPool() (privKey []byte, pubKey []byte, err error)
 
-	// GenerateHostCert generates host certificate, it takes pkey as a signing
-	// private key (host certificate authority)
-	GenerateHostCert(pkey, key []byte, hostID, authDomain string, roles teleport.Roles, ttl time.Duration) ([]byte, error)
+	// GenerateHostCert takes the private key of the CA, public key of the new host,
+	// along with metadata (host ID, node name, cluster name, roles, and ttl) and generates
+	// a host certificate.
+	GenerateHostCert(privateCASigningKey, publicHostKey []byte, hostID, nodeName, clusterName string, roles teleport.Roles, ttl time.Duration) ([]byte, error)
```

Comment:
there is a lot of parameters in this function, at this point it makes sense to introduce the CertParams struct and put fields there